app/utils/theme.js

```diff
@@ -27,3 +30,24 @@ export function setNavigatorStyles(navigator, theme) {
         screenBackgroundColor: theme.centerChannelBg,
     });
 }
+
+export function getAllowedThemes(state) {
```

Comment:
Since we're passing this as a prop to one of the components through `mapStateToProps`, we need to memoize it, otherwise the component that it's being passed to will re-render every time `mapStateToProps` is called (which is a lot). Easiest way to do that would be to use a [reselect selector function](https://github.com/reduxjs/reselect) which is basically a pure function that takes the state, gets the important parts of it, and uses those to calculate some final value while being smart enough to skip that when the important parts don't change. We have more info about them [here](https://developers.mattermost.com/contribute/redux/selectors/).

This one is fairly straightforwards since it just grabs the config and theme and then does some work on those, so we just need to pass them into `createSelector` first and then the final result function will just contain the rest of this function
```javascript
import {createSelector} from 'reselect';

export const getAllowedThemes = createSelector(
    getConfig,
    getTheme,
    (config, activeTheme) => {
        // everything else here
    }
);
```

Also, could you move it into something like `app/selectors/themes`? It would be nice to move it into mattermost-redux eventually if we end up using this on the web app as well, but that's unnecessary for now.